pkg/maps/set_example_test.go

```diff
@@ -0,0 +1,142 @@
+package maps_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"gopkg.in/typ.v3/pkg/maps"
+	"gopkg.in/typ.v3/pkg/sets"
+)
+
+func ExampleSet() {
+	set := make(maps.Set[string], 0)
+	set.Add("A")
+	set.Add("B")
+	set.Add("C")
+
+	set.Range(func(value string) bool {
+		fmt.Println("Value:", value)
+		return true
+	})
+
+	// Unordered output:
+	// Value: A
+	// Value: B
+	// Value: C
+}
+
+func ExampleSet_setOperations() {
+	set1 := make(maps.Set[string], 0)
+	set1.Add("A")
+	set1.Add("B")
+	set1.Add("C")
+	fmt.Println("set1:", set1) // {A B C}
+
+	set2 := make(maps.Set[string], 0)
+	set2.Add("B")
+	set2.Add("C")
+	set2.Add("D")
+	fmt.Println("set2:", set2) // {B C D}
+
+	fmt.Println("union:", set1.Union(set2))         // {A B C D}
+	fmt.Println("intersect:", set1.Intersect(set2)) // {B C}
+	fmt.Println("set diff:", set1.SetDiff(set2))    // {A}
+	fmt.Println("sym diff:", set1.SymDiff(set2))    // {A D}
+
+	// Please note: the Set.String() output is not sorted!
+}
+
+func TestSet_SetOperations(t *testing.T) {
+	setA := newSetA()
+	assert.ElementsMatch(t, setA.Slice(), []string{"A", "B", "C"}, "setA values")
```

Comment:
Any tip for built in version of elements match or should I just write my own?

Reply:
```go
want := "A, B, C"
got := strings.Join(setA.Slice(), ", ")
```

Reply:
Considering it's based on maps and is unordered, anything else? Not that it's a complicated thing to write for this case, but if there's something built in that'd be nice.

Reply:
Nothing built in, but you could write a function for it in `internal/assert`

Reply:
Such as:

```go
func ElementsMatch[S ~[]string](t *testing.T, want, got S) {
    want := slices.Clone(want)
    got := slices.Clone(got)
    slices.Sort(want)
    slices.Sort(got)
    wantStr := strings.Join(want, ", ")
    gotStr := strings.Join(got, ", ")
    if wantStr != gotStr {
        t.Errorf("want %q, got %q", wantStr, gotStr)
    }
}
```